ivy/functional/frontends/tensorflow/raw_ops.py

```diff
@@ -131,6 +131,12 @@ def Square(*, x, name="Square"):
     return ivy.square(x)
 
 
+def Softmax(*, x, axis=None, name="Softmax"):
+    if x.dtype not in ["half", "float32", "float64"]:
+        x = ivy.astype(x, "float64")
```

Comment:
I think that we should cast to float, but not necessarily float64 in all cases. We should maybe look at casting to the nearest reasonable float dtype instead